runtime/src/iree/runtime/demo/BUILD.bazel

```diff
@@ -64,12 +65,21 @@ iree_runtime_cc_binary(
     name = "hello_world_terse",
     srcs = ["hello_world_terse.c"],
     deps = [
+        ":simple_mul_module_c",
         "//runtime/src/iree/runtime",
-        "//runtime/src/iree/runtime/testdata:simple_mul_module_c",
     ],
 )
 
 native_test(
     name = "hello_world_terse_test",
     src = ":hello_world_terse",
 )
+
+iree_bytecode_module(
+    name = "simple_mul_module",
+    src = "simple_mul.mlir",
+    c_identifier = "iree_runtime_testdata_simple_mul_module",
```

Comment:
Those changes will help with the CI failures on https://github.com/iree-org/iree/actions/runs/9353849536/job/25745381896?pr=17540#step:4:113

(I'm now wondering why `build_all` succeeded... is that not building this demo?)

Reply:
ah i thought i caught those (even though they're in BUILD)

Reply:
I'd prefer to include `demo` in the name, so `iree_runtime_demo_simple_mul_module` instead of `iree_runtime_simple_mul_module` (that makes it seem like it's part of the public API and not something specific to a demo). Not too big a deal though.